RoddyCore/src/de/dkfz/roddy/knowledge/brawlworkflows/BrawlWorkflow.groovy

```diff
@@ -1,417 +1,199 @@
+        return GenericMethod.callGenericTool(toolName, input, additionalInput);
+    }
 
-        temp << loadFilesCall
-        classOfFileObject
+    @CompileDynamic
+    protected FileGroup runWithOutputFileGroup(String toolName, BaseFile input, int numericCount, Object... additionalInput) {
+        return GenericMethod.callGenericToolWithFileGroupOutput(toolName, input, numericCount, additionalInput);
     }
 
-    private static Type findTypeOfMethodOrProperty(Class _classOfCallingObject, String[] splitCall, int indexInSplit) {
-        Type classToLoad;
-        Method method = _classOfCallingObject.methods.find { Method m -> m.name == splitCall[indexInSplit].split("[(]")[0].replaceAll("[()]", "") }
+    @CompileDynamic
+    protected FileGroup runWithOutputFileGroup(String toolName, BaseFile input, List<String> indices, Object... additionalInput) {
+        return GenericMethod.callGenericToolWithFileGroupOutput(toolName, input, indices, additionalInput);
+    }
 
-        if (method) {
+    List<String> runSynchronized(String toolID, Map<String, Object> parameters) {
+        return ExecutionService.getInstance().callSynchronized(context, toolID, parameters);
+    }
 
-            classToLoad = method.genericReturnType;
-        }
-        Field field = _classOfCallingObject.fields.find { Field f -> f.name == splitCall[indexInSplit].split("[(]")[0].replaceAll("[()]", "") }
-        if (field) {
-            classToLoad = field.genericType;
-        }
+    protected boolean getflag(String flagID, boolean defaultValue = true) {
+        return context.getConfiguration().getConfigurationValues().getBoolean(flagID, defaultValue);
+    }
 
-        if (splitCall.size() == indexInSplit + 1)
-            return classToLoad;
-        else {
-            //Is the next in line a method or a field?...
-            //If the current classToLoad is some sort of generic object... then what?
-            // Unfortunately, Roddy uses generics at some points, e.g. for Tuples. However, Generics can be a huge pain.
-            // If our next call is a method, we will just go on, if it is a value and the current type is generic... Let's handle that differently.
-            boolean nextIsMethod = isNextMethodCall(method ? method.returnType : field.type, splitCall, indexInSplit + 1)
-            if (nextIsMethod || (!nextIsMethod && !classToLoad.typeName.contains("<")))
-                return findTypeOfMethodOrProperty(method ? method.returnType : field.type, splitCall, indexInSplit + 1);
-            else {
-                // Now handle the generic fields...
-                //Get the fields name first.
-                String fieldName = splitCall[indexInSplit + 1];
-                int index = method.getReturnType().fields.findIndexOf { Field it -> it.name == fieldName; };
+    protected BaseFile getSourceFile(String path, String _class = BaseFile.STANDARD_FILE_CLASS) {
```

Comment:
From an internal perspective source file makes sense. On the DSL level a simple `file` may be sufficient. Consider also the next comment (typed parameters plus dynamic dispatch instead of method-name-plus-untyped(string)-parameters).